monitor/internal/pod/monitor.go

```diff
@@ -196,47 +171,101 @@ func (m *PodMonitor) Resync(ctx context.Context) error {
 	return ResyncWithAllPods(ctx, m.kubeClient, m.resyncInfo, m.eventsCh, m.localNode)
 }
 
-func (m *PodMonitor) startManager(ctx context.Context, mgr manager.Manager) error {
-	controllerStarted := make(chan struct{})
-	if err := mgr.Add(&runnable{ch: controllerStarted}); err != nil {
-		return fmt.Errorf("pod: %s", err.Error())
-	}
+const (
+	startupWarningMessage = "pod: the Kubernetes controller did not start within the last 5s. Waiting..."
+)
+
+var (
+	retrySleep          = time.Second * 3
+	warningMessageSleep = time.Second * 5
+	managerNew          = manager.New
+)
+
+func (m *PodMonitor) startManager(ctx context.Context) {
+	var mgr manager.Manager
 
-	// starting the manager is a bit awkward:
-	// - it does not use contexts
-	// - we pass in a fake signal handler channel
-	// - we start another go routine which waits for the context to be cancelled
-	//   and closes that channel if that is the case
 	startTimestamp := time.Now()
 	z := make(chan struct{})
-	errCh := make(chan error, 2)
-	go func() {
-		<-ctx.Done()
-		close(z)
-		errCh <- ctx.Err()
-	}()
+	controllerStarted := make(chan struct{})
+
 	go func() {
-		if err := mgr.Start(z); err != nil {
-			errCh <- err
+		// manager.New already contacts the Kubernetes API
+		for {
```

Comment:
THis loop is just make sure we get the manager, right?

Reply:
exactly ... all errors must be handled here, that's why I also removed the error return